presto-main/src/main/resources/webapp/timeline.html

```diff
@@ -0,0 +1,132 @@
+<!DOCTYPE html>
+<head xmlns="http://www.w3.org/1999/html" xmlns="http://www.w3.org/1999/html">
+    <meta charset="utf-8">
+
+    <script src="vendor/d3/d3-3.3.4.js"></script>
+
+    <link href="vendor/bootstrap/css/bootstrap.css" rel="stylesheet" type="text/css">
+
+    <link rel="stylesheet" href="vendor/vis/dist/vis.min.css">
+    <script src="vendor/vis/dist/vis.min.js"></script>
+
+    <style type="text/css">
+        .vis.timeline .item {
+        padding: 0px;
+        }
+
+        .vis.timeline .labelset .vlabel .inner {
+        padding: 0px;
+        }
+
+        .vis.timeline .item.range {
+        height: 3px;
+        }
+
+        .vis.timeline .red {
+        background-color: #F2DEDE;
+        border-color: #F2AEAE;
+        }
+        .vis.timeline .green {
+        background-color: #DFF0DB;
+        border-color: #B8F0AA;
+        }
+        .vis.timeline .blue {
+        background-color: #E3E9FC;
+        border-color: #B0C3FC;
+        }
+    </style>
+</head>
+
+<body>
+
+
+<div class="container">
+    <div class="page-header">
+        <h1>Presto :: Query <span id="queryId"></span></h1>
+    </div>
+
+    <h2>Timeline</h2>
+
+    <div id="timeline"></div>
+</div>
+
+</body>
+
+<script>
+d3.json('/v1/query/' + window.location.search.substring(1), function (query) {
+    d3.select('#queryId').text(query.queryId);
+    renderTimeline(query);
+});
+
+function renderTimeline(data) {
+    function getTasks(stage) {
+        return [].concat.apply(
+                stage.tasks,
+                stage.subStages.map(getTasks));
+    }
+    tasks = getTasks(data.outputStage);
+    tasks = tasks.map(function(task) {
+        return {
+            taskId: task.taskId.substring(task.taskId.indexOf('.') + 1),
+            time: {
+                create: task.stats.createTime,
+                firstStart: task.stats.firstStartTime,
+                lastStart: task.stats.lastStartTime,
+                end: task.stats.endTime,
+            },
+        };
+    });
+
+    var groups = new vis.DataSet();
+    var items = new vis.DataSet();
+    for (var i = 0; i < tasks.length; i++) {
+        var task = tasks[i];
+        var taskIdMain = task.taskId.substr(0, task.taskId.indexOf("."));
+        var taskIdMinor = task.taskId.substr(task.taskId.indexOf(".")+1);
+        if (taskIdMinor == 0) {
+            groups.add({
+                id: taskIdMain,
+                content: task.taskIdMain,
```

Comment:
where was `task.taskIdMain` set? it looks like this is going to evaluate to `undefined`